lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -18,20 +18,23 @@ def install!
 
             UI.message "- Installing target `#{target.name}` #{target.platform}" do
               add_target
+              add_test_targets if target.contains_test_specifications?
               create_support_files_dir
               add_resources_bundle_targets
               add_files_to_build_phases
               create_xcconfig_file
+              create_test_xcconfig_files if target.contains_test_specifications?
               if target.requires_frameworks?
                 create_info_plist_file
                 create_module_map
                 create_umbrella_header do |generator|
+                  file_accessors = target.file_accessors.reject { |f| f.spec.test_specification? }
```

Comment:
This prevents adding test specification headers as part of the generated umbrella header